entrypoint.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+#
+# Check uid/gid of installation dir
+#
+
+set -e
+
+if [ ! -d $INSTALL_DEST ]; then
+    echo "Error: '$INSTALL_DEST' does not exists"
+    exit 1
+fi
+
+if [ "$(id -u)" = '0' ]; then
+    export LIZMAP_UID=$(stat -c '%u' $INSTALL_DEST)
+    export LIZMAP_GID=$(stat -c '%g' $INSTALL_DEST)
+    # Warn if the destination is owned by
+    # Root, this may indicates that the directory was created
+    # at binding
+    if [ "$LIZMAP_UID" = '0' ]; then
+        echo "WARNING: Your destination directory is owned by 'root'"
+    else
+        SUEXEC="su-exec $LIZMAP_UID:$LIZMAP_GID"
+    fi
+else
+    export LIZMAP_UID=$(id -u)
+    export LIZMAP_GID=$(id -g)
+fi
+
+#
+# Commands
+#
+
+_makedirs() {
+    $SUEXEC mkdir -p $INSTALL_DEST/plugins \
+             $INSTALL_DEST/processing \
+             $INSTALL_DEST/wps-data \
+             $INSTALL_DEST/www/var/log \
+             $INSTALL_DEST/var/log/nginx \
+             $INSTALL_DEST/var/nginx-cache \
+             $INSTALL_DEST/var/lizmap-theme-config \
+             $INSTALL_DEST/var/lizmap-db \
+             $INSTALL_DEST/var/lizmap-config \
+             $INSTALL_DEST/var/lizmap-modules \
+             $INSTALL_DEST/var/lizmap-my-packages
+}
+
+_makenv() {
+    source $INSTALL_SOURCE/env.default
+    LIZMAP_PROJECTS=${LIZMAP_PROJECTS:-"$LIZMAP_INSTALL_DIR/instances"} 
+    cat > $INSTALL_DEST/.env <<-EOF
+		LIZMAP_PROJECTS=$LIZMAP_PROJECTS
+		LIZMAP_DIR=$LIZMAP_INSTALL_DIR
+		LIZMAP_UID=$LIZMAP_UID
+		LIZMAP_GID=$LIZMAP_GID
+		LIZMAP_VERSION_TAG=$LIZMAP_VERSION_TAG
+		QGIS_VERSION_TAG=$QGIS_VERSION_TAG
+		POSTGIS_VERSION=$POSTGIS_VERSION
+		POSTGRES_PASSWORD=$POSTGRES_PASSWORD
+		QGIS_MAP_WORKERS=$QGIS_MAP_WORKERS
+		WPS_NUM_WORKERS=$WPS_NUM_WORKERS
+		LIZMAP_PORT=$LIZMAP_PORT
+		OWS_PORT=$OWS_PORT
+		WPS_PORT=$WPS_PORT
+		POSTGIS_PORT=$POSTGIS_PORT
+		POSTGIS_ALIAS=$POSTGIS_ALIAS   
+		EOF
+    chown $LIZMAP_UID:$LIZMAP_GID $INSTALL_DEST/.env
+}
+
+
+_install-plugin() {
+    local plugindir=$INSTALL_DEST/plugins/lizmap
+    if [ -d $plugindir ]; then
+        local version=$(cat $plugindir/metadata.txt | grep "version=" |  cut -d '=' -f2)
+        echo "Found installed version $version"
+        if [ ! "$LIZMAP_PLUGIN_VERSION" = "$version" ]; then
+            echo "Removing installed version"
+            rm -rf $plugindir
+        else
+            echo "Installed version match required version"
+            return 0
+        fi
+    fi
+    echo "Installing version $LIZMAP_PLUGIN_VERSION"
+    wget https://github.com/3liz/lizmap-plugin/releases/download/$LIZMAP_PLUGIN_VERSION/lizmap.$LIZMAP_PLUGIN_VERSION.zip \
```

Comment:
```suggestion
    echo "Installing version 'master'"
    wget https://packages.3liz.org/pub/lizmap-qgis-plugin/master/lizmap-qgis-plugin.master.zip \
```

And remove this variable ?

The `master` version is the stable branch, always ready to be released on http://plugins.qgis.org

Reply:
My concern is is to get reproductible installation, by using master:  1) You have to download the package each time you reconfigure (which may be a lot when testing) 2) I cannot ensure reproductibility since master may have changed between invocation.

LIZMAP_PLUGIN_VERSION may be configurable at installation, so it may be changed when upgrading - as installation parameter, it is persistent.